gui/wxpython/core/ws.py

```diff
@@ -170,14 +170,21 @@ def OnRenderDone(self, event):
         self.mapMerger.AddRasterBands(self.tempMap, {1: 1, 2: 2, 3: 3})
         del self.mapMerger
 
+        add_alpha_channel = True
+        mask_fill_value = 0
+        if self.fetching_cmd[1]['format'] == 'jpeg':
+            mask_fill_value = 256
```

Comment:
Thank you for fixing it, could you explain the fill value 256?

Reply:
You are welcome. Correct value should be 255 (white color), not 256 (but every value > 255  produce white color in this case). Alright I updated it to the correct value. If mask is white color `g.pnmcomp` doesn't apply mask for the input map (WMS jpeg source format) and rendering is correctly.

/gui/wxpython/core/render.py, row 658
```
if maps:
    ret, msg = RunCommand('g.pnmcomp',
                          getErrorMsg=True,
                          overwrite=True,
                          input='%s' % ",".join(maps),
                          mask='%s' % ",".join(masks),
                          opacity='%s' % ",".join(opacities),
                          bgcolor=bgcolor,
                          width=env['GRASS_RENDER_WIDTH'],
                          height=env['GRASS_RENDER_HEIGHT'],
                          output=self.Map.mapfile,
                          env=env)
```